hub-saml/src/test/java/uk/gov/ida/saml/hub/transformers/outbound/HubEidasAttributeQueryRequestToSamlAttributeQueryTransformerTest.java

```diff
@@ -0,0 +1,212 @@
+import static uk.gov.ida.saml.hub.domain.UserAccountCreationAttribute.DATE_OF_BIRTH;
+import static uk.gov.ida.saml.hub.test.builders.HubEidasAttributeQueryRequestBuilder.aHubEidasAttributeQueryRequest;
+
+@RunWith(OpenSAMLMockitoRunner.class)
+public class HubEidasAttributeQueryRequestToSamlAttributeQueryTransformerTest {
+
+    public static final String ENCRYPTED_IDENTITY_ASSERTION = "encrypted-identity-assertion!";
+    public static final String AUTHN_STATEMENT_ID = "AUTHEN_STATEMENT_ID";
+
+    private HubEidasAttributeQueryRequestToSamlAttributeQueryTransformer transformer;
+
+    private OpenSamlXmlObjectFactory openSamlXmlObjectFactory;
+
+    @Mock
+    private IdaStatusMarshaller<?> statusMarshaller;
+
+    @Mock
+    private AttributeFactory_1_1 attributeFactory;
+
+    @Mock
+    private StringToOpenSamlObjectTransformer<Assertion> stringAssertionTransformer;
+
+    @Mock
+    private OutboundAssertionToSubjectTransformer outboundAssertionToSubjectTransformer;
+
+    @Mock
+    private AttributeQueryAttributeFactory attributeQueryAttributeFactory;
+
+    @Mock
+    private EncryptedAssertionUnmarshaller encryptedAssertionUnmarshaller;
+
+    @Before
+    public void setup() throws Exception {
+        openSamlXmlObjectFactory = new OpenSamlXmlObjectFactory();
+        HubAssertionMarshaller assertionTransformer = new HubAssertionMarshaller(openSamlXmlObjectFactory, attributeFactory, outboundAssertionToSubjectTransformer);
+        AssertionFromIdpToAssertionTransformer assertionFromIdpAssertionTransformer = new AssertionFromIdpToAssertionTransformer(stringAssertionTransformer);
+
+        when(stringAssertionTransformer.apply(anyString())).thenReturn(anAssertion().buildUnencrypted());
+
+        transformer = new HubEidasAttributeQueryRequestToSamlAttributeQueryTransformer(
+            openSamlXmlObjectFactory,
+            assertionTransformer,
+            assertionFromIdpAssertionTransformer,
+            attributeQueryAttributeFactory,
+            encryptedAssertionUnmarshaller);
+    }
+
+    @Test
+    public void shouldTransformProperly() throws Exception {
+        PersistentId persistentId = aPersistentId().build();
+        HubEidasAttributeQueryRequest hubEidasAttributeQueryRequest = aHubEidasAttributeQueryRequest()
+            .withId("originalId")
+            .withPersistentId(persistentId)
+            .build();
+
+        AttributeQuery transformedQuery = transformer.apply(hubEidasAttributeQueryRequest);
+
+        assertThat(transformedQuery.getID()).isEqualTo(hubEidasAttributeQueryRequest.getId());
+        assertThat(transformedQuery.getSubject().getNameID().getValue()).isEqualTo(persistentId.getNameId());
+        assertThat(transformedQuery.getIssuer().getValue()).isEqualTo(hubEidasAttributeQueryRequest.getIssuer());
+        assertThat(transformedQuery.getVersion()).isEqualTo(SAMLVersion.VERSION_20);
+        new XmlObjectToBase64EncodedStringTransformer<>().apply(transformedQuery);
```

Comment:
What is this line doing? My only guess is that it asserts that it is valid xml but if so, I would prefer for this to be explicit

Reply:
I cloned `HubAttributeQueryRequestToSamlAttributeQueryTransformerTest.java` to create `HubAttributeQueryRequestToSamlAttributeQueryTransformerTest.java`. This line should not be there. I have removed it.